### PR TITLE
Azure monitor: Basic Logs frontend

### PR DESCRIPTION
**Pull request by jcolladokuri**
**What is this feature?**

Updates Azure Monitor frontend to support Basic Logs

- Adds toggle on Datasource Level to enable / disable Basic Logs for the datasource
- Adds toggle on Panel level to switch between Analytics and Basic

Datasource config:

image

Panel level toggle:

image

The above only shows if:
1. Selected resource is Log Analytics workspace
2. Basic Logs has been enabled on the datasource
3. Query is not an alerts query

image

Upcoming frontend work:
1. Modal to acknowledge per query cost when toggling from Analytics to basic
2. Usage notification display when query is run

Fixes #
#84946 
#84947

### CODE REVIEW

File: public/app/plugins/datasource/azuremonitor/components/ConfigEditor/BasicLogsToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        '&:hover': {
+          textDecoration: 'none',
+        },
+      }),
+    }),
+  };
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => onBasicLogsEnabledChange(e.target.checked);
+  const description = (
+    <p className={styles.text}>
+      Enabling this feature incurs Azure Monitor per-query cost on dashboard panels that use tables configured for{' '}
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/ConfigEditor/BasicLogsToggle.tsx:34]:
```suggestion
      Enabling this feature incurs Azure Monitor per-query costs on dashboard panels that query tables configured for{' '}
```

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsQueryEditor.tsx [hunk 2]
```diff
@@ -42,6 +45,8 @@ const LogsQueryEditor = ({
   data,
 }: LogsQueryEditorProps) => {
   const migrationError = useMigrations(datasource, query, onChange);
+  const [showBasicLogsToggle, setshowBasicLogsToggle] = useState<boolean>(false);
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsQueryEditor.tsx:48]:
```suggestion
  const [showBasicLogsToggle, setShowBasicLogsToggle] = useState<boolean>(false);
```

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsQueryEditor.tsx [hunk 3]
```diff
@@ -65,6 +70,20 @@ const LogsQueryEditor = ({
     }
   }, [query.azureLogAnalytics?.resources, datasource.azureLogAnalyticsDatasource]);
 
+  useEffect(() => {
+    if (query.azureLogAnalytics?.resources && query.azureLogAnalytics.resources.length === 1) {
+      const resource = parseResourceURI(query.azureLogAnalytics.resources[0]);
+      setshowBasicLogsToggle(
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsQueryEditor.tsx:76]:
```suggestion
      setShowBasicLogsToggle(
```

**Review by aangelisc — APPROVED** [review]
Great work on this @jcolladokuri! This all LGTM, just a couple nits in the comments

**Review by aangelisc — COMMENTED** [review]
There are some warnings in the `LogsQueryEditor` and `TimeManagement` components warning about missing dependencies in some `useEffect` hooks that'll need to be fixed before this is merged 😊

**Comment by jcolladokuri** [comment]
Thanks @aangelisc! I have made the updates accordingly, ended up changing the logic a bit to avoid unnecessary `onchange` calls after updating the dependency lists. Let me know your thoughts!

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/TimeManagement.tsx [hunk 4]
```diff
@@ -80,7 +80,6 @@ export function TimeManagement({ query, onQueryChange: onChange, schema }: Azure
 
   useEffect(() => {
     if (query.azureLogAnalytics?.basicLogsQuery) {
-      onChange(setDashboardTime(query, 'dashboard'));
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/TimeManagement.tsx]: I think you still need this line (it's causing a failing test)

**Review by aangelisc — COMMENTED** [review]
Thanks @jcolladokuri, you just need to fix your failing test and then we should be able to approve this for merging 😊

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/TimeManagement.tsx (hunk 4 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/TimeManagement.tsx]: Having this here was causing an infinite loop amongst the dependency updates so I have this handled by the `LogsManagement` component, I have updated the tests both here and for `LogsManagement` to make sure this is accounted for.

**Comment by aangelisc** [comment]
Going ahead and merging this, the build failures aren't related to this PR 😊 thanks @jcolladokuri!

**Pull request merged**